dangerzone/isolation_provider/qubes.py

```diff
@@ -77,6 +77,32 @@ def start_doc_to_pixels_proc(self) -> subprocess.Popen:
 
         return p
 
+    def terminate_doc_to_pixels_proc(
+        self, document: Document, p: subprocess.Popen
+    ) -> None:
+        """Terminate a spawned disposable qube.
+
+        Qubes does not offer a way out of the box to terminate disposable Qubes from
+        domX [1]. Our best bet is to close the standard streams of the process, and hope
+        that the disposable qube will attempt to read/write to them, and thus receive an
+        EOF.
+
+        There are two ways we can do the above; close the standard streams explicitly,
+        or terminate the process. The problem with the latter is that terminating
+        `qrexec-client-vm` happens immediately, and we no longer have a way to learn if
+        the disposable qube actually terminated. That's why we prefer closing the
+        standard streams explicitly, so that we can afterwards use `Popen.wait()` to
+        learn if the qube terminated.
```

Comment:
And yet we've seen in our testing that if the process in the VM does not block on the file descriptor, it will not be killed. I can reproduce this by spawning the `qubes.VMShell` service in a disposable qube, running `sleep`, and then killing the process.

So, I bet we're missing something here. We should probably investigate this together at some point. In any case, the code in this PR will ultimately kill the process, after closing the file descriptors, so it covers that case as well.